projects/dsp-ui/src/lib/viewer/views/resource-view/resource-view.component.ts

```diff
@@ -240,6 +251,64 @@ export class ResourceViewComponent implements OnInit, OnChanges, OnDestroy {
         }
     }
 
+    /**
+     * Updates the standoff link value for the resource being displayed.
+     *
+     */
+    private _updateStandoffLinkValue() {
+
+        if (this.resource === undefined) {
+            return;
+        }
+
+        const gravsearchQuery = `
+ PREFIX knora-api: <http://api.knora.org/ontology/knora-api/simple/v2#>
+ CONSTRUCT {
+     ?res knora-api:isMainResource true .
+     ?res knora-api:hasStandoffLinkTo ?target .
+ } WHERE {
+     BIND(<${this.resource.id}> as ?res) .
+     OPTIONAL {
+         ?res knora-api:hasStandoffLinkTo ?target .
+     }
+ }
+ OFFSET 0
+ `;
+
+        this._dspApiConnection.v2.search.doExtendedSearch(gravsearchQuery).subscribe(
+            (res: ReadResourceSequence) => {
+
+                // one resource is expected
+                if (res.resources.length !== 1) {
+                    return;
```

Comment:
this should never happen since the `Optional` is used: also if there is no standoff link value, the resource should be returned. However, being paranoid `res.resources[0]` without an assertion seems dangerous to me.

Reply:
let's discss the whole flow of data together today or tomorrow to make sure I did not make an error in reasoning.